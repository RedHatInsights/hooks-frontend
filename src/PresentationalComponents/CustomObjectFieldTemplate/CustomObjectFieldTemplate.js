import React from 'react';
import {
    Title
} from '@patternfly/react-core';
import PropTypes from 'prop-types';

export const CustomObjectFieldTemplate = ({ title, description, properties }) => {
    return (
        <div>
            <Title size='md'>{ title }</Title>
            { description }
            { properties.map(element => (
                element.content
            )) }
        </div>
    );
};

CustomObjectFieldTemplate.propTypes = {
    title: PropTypes.string,
    description: PropTypes.string,
    properties: PropTypes.oneOfType([ PropTypes.object, PropTypes.array ]),
    name: PropTypes.string
};
