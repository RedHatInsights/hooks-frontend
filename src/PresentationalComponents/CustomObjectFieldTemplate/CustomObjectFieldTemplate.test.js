import toJson from 'enzyme-to-json';

import { CustomObjectFieldTemplate } from './CustomObjectFieldTemplate';

describe('CustomObjectFieldTemplate', () => {
    it('expect to render', () => {
        const testProps = {
            title: 'Test Object',
            description: 'This is a test',
            name: 'test-object',
            properties: [
                { content: 'Test Content' }
            ]
        };
        const wrapper = shallow(
            <CustomObjectFieldTemplate { ...testProps } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
