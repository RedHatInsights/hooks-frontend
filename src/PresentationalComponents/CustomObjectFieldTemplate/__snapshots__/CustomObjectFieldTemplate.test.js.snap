// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomObjectFieldTemplate expect to render 1`] = `
<div>
  Test Object
  This is a test
  Test Content
</div>
`;
